src/Kuzzle.js

```diff
@@ -29,28 +28,23 @@ const
 class Kuzzle extends KuzzleEventEmitter {
 
   /**
-   * @param network - the network wrapper to use. if the argument is a string, creates an embeed network wrapper if a
-   * @param [options] - Connection options
+   * @param protocol - the protocol to use
+   * @param [options] - Kuzzle options
    */
-  constructor(network, options = {}) {
+  constructor(protocol, options = {}) {
     super();
 
-    if (network === undefined || network === null) {
-      throw new Error('"network" argument missing');
+    if (protocol === undefined || protocol === null) {
+      throw new Error('"protocol" argument missing');
     }
 
-    // embedded network protocol (http/websocket/socketio):
-    if (typeof network === 'string') {
-      return new Kuzzle(networkWrapper(network, options), options);
-    }
-
-    // custom protocol: check the existence of required methods
+    // check the existence of required methods
     for (const method of ['addListener', 'isReady', 'query']) {
-      if (typeof network[method] !== 'function') {
-        throw new Error(`Network instance must implement a "${method}" method`);
+      if (typeof protocol[method] !== 'function') {
```

Comment:
Shouldn't we check that the provided protocol is an instance of `KuzzleAbstractProtocol` instead?